File: java_vs_python/testme.py
from __future__ import annotations
from typing import List
import time
FLOATING_POINT_ACCURACY = 1.0e-6

X = 0
Y = 1
Z = 2

p1 = (0,0,0)
p2 = (1,1,1)
p3 = (0,2,0)

def triangleAreaXY(p1:tuple[float,float,float], p2:tuple[float,float,float], p3:tuple[float,float,float]) :
    
    # p1x = p1[X]
    # p1y = p1[Y]
    # p2x = p2[X]
    # p2y = p2[Y]
    # p3x = p3[X]
    # p3y = p3[Y]
    a = abs(p1[X]*(p2[Y]-p3[Y]) +p2[X]*(p3[Y]-p1[Y]) + p3[X]*(p1[Y]-p2[Y]))/2.0
    return a


#   //if include_edge is false : if the point is on the edge, is NOT inside. 
#   //if include_edge is true  : if the point is on the edge, is inside. 
def insideTriangleXY(p:tuple[float,float,float], include_edge : bool) -> bool :
    area  = triangleAreaXY(p1, p2, p3)
    area1 = triangleAreaXY(p, p2, p3)
    area2 = triangleAreaXY(p1, p, p3)
    area3 = triangleAreaXY(p1, p2, p)
    if not include_edge :
        if (area1 <= FLOATING_POINT_ACCURACY) or (area2 <= FLOATING_POINT_ACCURACY) or (area3 <= FLOATING_POINT_ACCURACY) :
            return False
    ins = (abs(area1+area2+area3) - area) <= (FLOATING_POINT_ACCURACY * area)
    return ins


if __name__ == '__main__':
    start_ms = time.perf_counter()

    a = (0,1,2)
    for i in range(int(1e6)):
        insideTriangleXY(a, True )

    end_ms = time.perf_counter()
    print(f"duration : {end_ms - start_ms}")

File: java_vs_python/testme.java
import java.util.Vector;
import java.lang.Math;

public class testme {

    float p1[], p2[], p3[];  // plane : nx*x + ny*y + nz*z = o
  
    testme() {
        p1 = new float[3];
        p1[0] = 0;
        p1[1] = 0;
        p1[2] = 0;
        p2 = new float[3];
        p2[0] = 1;
        p2[1] = 1;
        p2[2] = 1;
        p3 = new float[3];
        p3[0] = 0;
        p3[1] = 2;
        p3[2] = 0;

        float a[] = new float[3];
        a[0] = 0;
        a[1] = 2;
        a[2] = 3;
        for (int i = 0; i < 1e6; i++){
            insideTriangleXY(a, true );
        }


    }

	public static void main(String[] args) {
		System.out.println("Hello desde el paquete consoletest");

        long start_ms = System.currentTimeMillis();

        testme tm = new testme();

        long finish_ms = System.currentTimeMillis();
        long timeElapsed_ms = finish_ms - start_ms;

		System.out.printf("tijd genomen ms %d ", timeElapsed_ms);
	}

    float triangleAreaXY(float p1[], float p2[], float p3[]) {
        return (float)Math.abs((p1[0]*(p2[1]-p3[1]) + p2[0]*(p3[1]-p1[1])+ p3[0]*(p1[1]-p2[1]))/2.0);
      }
    
      //if include_edge is false : if the point is on the edge, is NOT inside. 
      //if include_edge is true  : if the point is on the edge, is inside. 
      boolean insideTriangleXY(float p[], boolean include_edge) {
        float FLOATING_POINT_ACCURACY = (float)1.0e-6;
        float area = triangleAreaXY(p1, p2, p3);
        float area1 = triangleAreaXY(p, p2, p3);
        float area2 = triangleAreaXY(p1, p, p3);
        float area3 = triangleAreaXY(p1, p2, p);
        if (!include_edge) {
          if ((area1 <= FLOATING_POINT_ACCURACY) || (area2 <= FLOATING_POINT_ACCURACY) || (area3 <= FLOATING_POINT_ACCURACY))
          {
            return false;
          }
        }
        return (( (float)Math.abs(area1+area2+area3) - area) <= FLOATING_POINT_ACCURACY * area);
      }
}

